app/usecases/tasks.js

```diff
@@ -1,41 +1,86 @@
-const msTodo = require("../services/todo")();
 
-module.exports = function() {
+module.exports = function(db, oAuth2Client) {
+  const msTodo = require("../services/todo")(db);
   const preferences = require("../services/preferences")(db);
+  const cal = require("../services/gcalendar")(db, oAuth2Client);
+  const gsearch = require("../services/gsearch");
 
   this.onUpdate = (ctx, waRes)=>{
     if (waRes.generic[0].text === "tasks_todo_auth") {
       msTodo.authorizeUser(ctx);
     } else if (waRes.generic[0].text === "tasks_show") {
       msTodo.getTodos().then((todos)=>{
+        let today = new Date();
+        today.setHours(9);
+        today.setMinutes(0);
+        today.setSeconds(0);
+        if (today.getTime() <= new Date().getTime()) {
+          today = new Date();
+        }
+        const tomorrow = new Date(today);
+        tomorrow.setDate(tomorrow.getDate() + 1);
```

Comment:
was machst du damit genau?

Reply:
Das ist für das Zeitfenster was ich für die is-busy funktion brauche. Ich mache ein Datum für heute 9 Uhr (damit niemand der um 1 uhr nachts beispielsweise nachschaut vom bot gesagt bekommt "hey du hast um 2 uhr zeit zum lernen"). Und falls es schon nach 9 uhr ist, dann setze ich das startDatum auf jetzt. Das Ende ist dann tomorrow, also die jetzige Zeit +24h.

Reply:
achso ok